### PR TITLE
[v2.9]  bump rancher machine to v0.15.0-rancher123

### PR DESCRIPTION
**Pull request by jiaqiluo**
https://github.com/rancher/rancher/issues/47257
https://github.com/rancher/rancher/issues/47644

<details>
<summary>Outdated Info</summary>

**Blocker**

The unit test failed with the following kind of error:
```
FAILgithub.com/rancher/rancher/pkg/auth/requests [build failed]
```
ref: https://github.com/rancher/rancher/actions/runs/11508369053/job/32038061423?pr=47700#step:3:455

To fix it, we need to backport the changes from [this PR](https://github.com/rancher/rancher/pull/47128)
 

</details>

### CODE REVIEW

File: pkg/apis/go.mod [hunk 1]
```diff
@@ -36,24 +36,24 @@ require (
 	github.com/rancher/gke-operator v1.9.3
 	github.com/rancher/norman v0.0.0-20240708202514-a0127673d1b9
 	github.com/rancher/rke v1.6.3
-	github.com/rancher/wrangler/v3 v3.0.0
+	github.com/rancher/wrangler/v3 v3.0.1-rc.2
```
Comment (jiaqiluo) [inline pkg/apis/go.mod:39]: Note to reviewers:
We need to match the version wit the one in the root `go.mod` file to make the CI happy.

File: go.mod [hunk 2]
```diff
@@ -164,21 +167,21 @@ require (
...
 	gopkg.in/yaml.v3 v3.0.1
 	helm.sh/helm/v3 v3.15.1
-	k8s.io/api v0.30.2
-	k8s.io/apiextensions-apiserver v0.30.1
-	k8s.io/apimachinery v0.30.2
-	k8s.io/apiserver v0.30.1
+	k8s.io/api v0.31.1
+	k8s.io/apiextensions-apiserver v0.31.1
+	k8s.io/apimachinery v0.31.1
+	k8s.io/apiserver v0.31.1
```
Comment (jiaqiluo) [inline go.mod:173]: Notes to reviewers: 
The change is generated by the `go mod tidy` command, the correct versions are pinged under the `replace` section.

File: pkg/apis/go.mod [hunk 3]
```diff
@@ -36,24 +36,24 @@ require (
...
 	github.com/sirupsen/logrus v1.9.3
-	k8s.io/api v0.30.2
-	k8s.io/apimachinery v0.30.2
+	k8s.io/api v0.31.1
+	k8s.io/apimachinery v0.31.1
```
Comment (jiaqiluo) [inline pkg/apis/go.mod:42]: Note to reviewers:
The change is generated by the go mod tidy command, the correct versions are pinged under the replace section.

File: go.mod [hunk 4]
```diff
@@ -8,6 +8,7 @@ replace (
 	github.com/containerd/containerd => github.com/containerd/containerd v1.6.27 // for compatibilty with docker 20.10.x
 	github.com/docker/distribution => github.com/docker/distribution v2.8.2+incompatible // rancher-machine requires a replace is set
 	github.com/docker/docker => github.com/docker/docker v20.10.27+incompatible // rancher-machine requires a repalce is set
+	github.com/google/cel-go => github.com/google/cel-go v0.17.8 // rancher-machine requires a repalce is set
```
Comment (snasovich) [inline go.mod:11]: Minor typo
```suggestion
	github.com/docker/docker => github.com/docker/docker v20.10.27+incompatible // rancher-machine requires a replace is set
	github.com/google/cel-go => github.com/google/cel-go v0.17.8 // rancher-machine requires a replace is set
```

Comment (jiaqiluo) [reply go.mod]: fixed

**Comment by snasovich** [comment]
Delaying this PR due to some versioning changes for `rancher/wrangler/v3` - `rancher/machine` version currently references `v3.0.1-rc2` while some upcoming bumps (e.g. https://github.com/rancher/rancher/pull/47753) in `release/v2.9` will be getting it to `v3.0.1-rc4`. This looks fine on the surface; however due to versioning changes for `rancher/wranger` `v3.0.1-rc4` is actually the same as `v3.0.1-rc1` (so `v3.0.1-rc4` is actually a "downgrade" compared to `v3.0.1-rc2`).

**Pull request merged**